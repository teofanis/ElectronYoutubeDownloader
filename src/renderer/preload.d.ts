import { IpcRequest, PromisedIpcResponse } from 'interfaces';
import { Channels } from 'main/preload';

declare global {
  interface Window {
    electron: {
      store: {
        getState: <T>() => T;
        setState: <T>(
          recipe: (draft: T) => void,
          description?: string | undefined
        ) => T;
        subscribe: <T>(
          listener: (state: T, description?: string | undefined) => void
        ) => () => void;
      };
      ipc: {
        send: <T>(
          channel: string,
          request: IpcRequest
        ) => PromisedIpcResponse<T>;
      };

      ipcRenderer: {
        removeAllListeners(channel?: Channels | string | null): void;
        sendMessage(channel: Channels, args: unknown[]): void;
        on(
          channel: string,
          func: (...args: unknown[]) => void
        ): (() => void) | undefined;
        once(channel: string, func: (...args: unknown[]) => void): void;
      };
    };
  }
}

export {};
