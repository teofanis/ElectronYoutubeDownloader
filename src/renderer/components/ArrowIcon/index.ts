// eslint-disable-next-line import/prefer-default-export
export { default as ArrowIcon } from './ArrowIcon';
