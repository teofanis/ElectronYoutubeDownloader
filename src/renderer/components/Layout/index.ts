/* eslint-disable import/prefer-default-export */
export { default as Layout } from './Layout';
