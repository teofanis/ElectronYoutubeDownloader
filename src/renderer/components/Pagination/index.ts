// eslint-disable-next-line import/prefer-default-export
export { default as Pagination } from './Pagination';
