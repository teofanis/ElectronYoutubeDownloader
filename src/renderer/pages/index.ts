import { Home } from './Home';

// eslint-disable-next-line import/prefer-default-export
export { Home };
