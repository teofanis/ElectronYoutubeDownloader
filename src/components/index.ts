// eslint-disable-next-line import/prefer-default-export
import { Button } from 'components/Button';
import { CancelButton } from 'components/CancelButton';
import { DownloadableItem } from 'components/DownloadableItem';
import { DownloadableItemControls } from 'components/DownloadableItemControls';
import { DownloadableItemTransition } from 'components/DownloadableItemTransition';
import { DownloadableItemTransitionContainer } from 'components/DownloadableItemTransitionContainer';
import { DownloadButton } from 'components/DownloadButton';
import { Downloader } from 'components/Downloader';
import { DownloadItemControl } from 'components/DownloadItemControl';
import { FileInput } from 'components/FileInput';
import { Heading } from 'components/Heading';
import { InputError } from 'components/InputError';
import { Layout } from 'components/Layout';
import { Logo } from 'components/Logo';
import { ProgressBar } from 'components/ProgressBar';
import { TextInput } from 'components/TextInput';
import { DownloadableItemTransition } from 'components/DownloadableItemTransition';
// Automatic import of components
export {
  Layout,
  Heading,
  Logo,
  Downloader,
  TextInput,
  FileInput,
  Button,
  DownloadButton,
  InputError,
  ProgressBar,
  CancelButton,
  DownloadableItem,
  DownloadableItemControls,
  DownloadItemControl,
  DownloadableItemTransitionContainer,
  DownloadableItemTransition,
  DownloadableItemTransition,
  // Automatic export of components
};
