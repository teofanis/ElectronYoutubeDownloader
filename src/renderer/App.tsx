import { Home } from 'pages';
import { MemoryRouter as Router, Route, Routes } from 'react-router-dom';
import './global.css';

export default function App() {
  return (
    <Router>
      <Routes>
        <Route path="/" element={<Home />} />
      </Routes>
    </Router>
  );
}
