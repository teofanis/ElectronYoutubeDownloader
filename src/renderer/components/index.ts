// eslint-disable-next-line import/prefer-default-export
import { Button } from 'renderer/components/Button';
import { CancelButton } from 'renderer/components/CancelButton';
import { DownloadableItem } from 'renderer/components/DownloadableItem';
import { DownloadableItemControls } from 'renderer/components/DownloadableItemControls';
import { DownloadableItemTransition } from 'renderer/components/DownloadableItemTransition';
import { DownloadableItemTransitionContainer } from 'renderer/components/DownloadableItemTransitionContainer';
import { DownloadButton } from 'renderer/components/DownloadButton';
import { Downloader } from 'renderer/components/Downloader';
import { DownloaderControls } from 'renderer/components/DownloaderControls';
import { DownloadItemControl } from 'renderer/components/DownloadItemControl';
import { FileInput } from 'renderer/components/FileInput';
import { Heading } from 'renderer/components/Heading';
import { InputError } from 'renderer/components/InputError';
import { Layout } from 'renderer/components/Layout';
import { Logo } from 'renderer/components/Logo';
import { ProgressBar } from 'renderer/components/ProgressBar';
import { TextInput } from 'renderer/components/TextInput';
import { Pagination } from 'renderer/components/Pagination';
// Automatic import of components
export {
  Layout,
  Heading,
  Logo,
  Downloader,
  TextInput,
  FileInput,
  Button,
  DownloadButton,
  InputError,
  ProgressBar,
  CancelButton,
  DownloadableItem,
  DownloadableItemControls,
  DownloadItemControl,
  DownloadableItemTransitionContainer,
  DownloadableItemTransition,
  DownloaderControls,
  Pagination,
  // Automatic export of components
};
