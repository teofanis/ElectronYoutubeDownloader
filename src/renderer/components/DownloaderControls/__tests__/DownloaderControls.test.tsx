import '@testing-library/jest-dom';
import { fireEvent, render } from '@testing-library/react';
import { DownloadQueue } from 'interfaces';
import { DownloaderControls } from 'renderer/components';

describe('DownloaderControls Test', () => {
  it('should render correctly', () => {
    expect(
      render(
        <DownloaderControls
          disableDownloadButton={false}
          downloadClickHandler={jest.fn}
          clearClickHandler={jest.fn}
          cancelClickHandler={jest.fn}
          downloadQueue={[]}
        />
      )
    ).toMatchSnapshot();
  });

  it('should render correctly with downloadQueue 1', () => {
    const downloadQueue = [
      {
        status: 'idle',
        url: 'https://www.youtube.com/watch?v=1',
      },
      {
        status: 'downloading',
        url: 'https://www.youtube.com/watch?v=2',
      },
      {
        status: 'enqueued',
        url: 'https://www.youtube.com/watch?v=3',
      },
    ] as DownloadQueue;

    const { container } = render(
      <DownloaderControls
        disableDownloadButton={false}
        downloadClickHandler={jest.fn}
        clearClickHandler={jest.fn}
        cancelClickHandler={jest.fn}
        downloadQueue={downloadQueue}
      />
    );
    expect(container.querySelectorAll('button')).toHaveLength(1);
    expect(container.querySelector('button')).toHaveTextContent('Cancel (2)');
  });

  it('should render correctly with downloadQueue 2', () => {
    const downloadQueue = [
      {
        status: 'cancelled',
        url: 'https://www.youtube.com/watch?v=1',
      },
      {
        status: 'downloading',
        url: 'https://www.youtube.com/watch?v=2',
      },
      {
        status: 'enqueued',
        url: 'https://www.youtube.com/watch?v=3',
      },
    ] as DownloadQueue;

    const { container } = render(
      <DownloaderControls
        disableDownloadButton={false}
        downloadClickHandler={jest.fn}
        clearClickHandler={jest.fn}
        cancelClickHandler={jest.fn}
        downloadQueue={downloadQueue}
      />
    );
    expect(container.querySelectorAll('button')).toHaveLength(1);
    expect(container.querySelector('button')).toHaveTextContent('Cancel');
  });

  it('should render correctly with downloadQueue 3', () => {
    const downloadQueue = [
      {
        status: 'cancelled',
        url: 'https://www.youtube.com/watch?v=1',
      },
      {
        status: 'idle',
        url: 'https://www.youtube.com/watch?v=2',
      },
      {
        status: 'errored',
        url: 'https://www.youtube.com/watch?v=3',
      },
    ] as DownloadQueue;

    const { container } = render(
      <DownloaderControls
        disableDownloadButton={false}
        downloadClickHandler={jest.fn}
        clearClickHandler={jest.fn}
        cancelClickHandler={jest.fn}
        downloadQueue={downloadQueue}
      />
    );
    expect(container.querySelectorAll('button')).toHaveLength(1);
    expect(container.querySelector('button')).toHaveTextContent('Clear (3)');
  });

  it('should render correctly with downloadQueue 4', () => {
    const downloadQueue = [
      {
        status: 'idle',
        url: 'https://www.youtube.com/watch?v=1',
      },
      {
        status: 'idle',
        url: 'https://www.youtube.com/watch?v=2',
      },
      {
        status: 'idle',
        url: 'https://www.youtube.com/watch?v=3',
      },
      {
        status: 'idle',
        url: 'https://www.youtube.com/watch?v=3',
      },
      {
        status: 'idle',
        url: 'https://www.youtube.com/watch?v=3',
      },
    ] as DownloadQueue;

    const { container } = render(
      <DownloaderControls
        disableDownloadButton={false}
        downloadClickHandler={jest.fn}
        clearClickHandler={jest.fn}
        cancelClickHandler={jest.fn}
        downloadQueue={downloadQueue}
      />
    );
    expect(container.querySelectorAll('button')).toHaveLength(1);
    expect(container.querySelector('button')).toHaveTextContent('Download (5)');
  });

  it('should render correctly with downloadQueue 5', () => {
    const downloadQueue = [
      {
        status: 'cancelled',
        url: 'https://www.youtube.com/watch?v=1',
      },
    ] as DownloadQueue;

    const { container } = render(
      <DownloaderControls
        disableDownloadButton={false}
        downloadClickHandler={jest.fn}
        clearClickHandler={jest.fn}
        cancelClickHandler={jest.fn}
        downloadQueue={downloadQueue}
      />
    );
    expect(container.querySelectorAll('button')).toHaveLength(1);
    expect(container.querySelector('button')).toHaveTextContent('Clear');
  });

  it('should render correctly with downloadQueue 6', () => {
    const downloadQueue = [
      {
        status: 'idle',
        url: 'https://www.youtube.com/watch?v=1',
      },
    ] as DownloadQueue;

    const { container } = render(
      <DownloaderControls
        disableDownloadButton={false}
        downloadClickHandler={jest.fn}
        clearClickHandler={jest.fn}
        cancelClickHandler={jest.fn}
        downloadQueue={downloadQueue}
      />
    );
    expect(container.querySelectorAll('button')).toHaveLength(1);
    expect(container.querySelector('button')).toHaveTextContent('Download');
  });

  it('should render correctly with downloadQueue 7', () => {
    const { container } = render(
      <DownloaderControls
        disableDownloadButton={false}
        downloadClickHandler={jest.fn}
        clearClickHandler={jest.fn}
        cancelClickHandler={jest.fn}
        downloadQueue={[]}
      />
    );
    expect(container.querySelectorAll('button')).toHaveLength(1);
    expect(container.querySelector('button')).toHaveTextContent('Download');
  });

  it('can handle clicks correctly', () => {
    const downloadClickHandler = jest.fn();
    const clearClickHandler = jest.fn();
    const cancelClickHandler = jest.fn();

    let { container } = render(
      <DownloaderControls
        disableDownloadButton={false}
        downloadClickHandler={downloadClickHandler}
        clearClickHandler={clearClickHandler}
        cancelClickHandler={cancelClickHandler}
        downloadQueue={[
          {
            status: 'idle',
            url: 'https://www.youtube.com/watch?v=1',
          },
        ]}
      />
    );

    let actionButton = container.querySelector('button') as HTMLButtonElement;
    expect(actionButton).toHaveTextContent('Download');
    fireEvent.click(actionButton);
    expect(downloadClickHandler).toHaveBeenCalledTimes(1);

    container = render(
      <DownloaderControls
        disableDownloadButton={false}
        downloadClickHandler={downloadClickHandler}
        clearClickHandler={clearClickHandler}
        cancelClickHandler={cancelClickHandler}
        downloadQueue={[
          {
            status: 'downloading',
            url: 'https://www.youtube.com/watch?v=1',
          },
        ]}
      />
    ).container;

    actionButton = container.querySelector('button') as HTMLButtonElement;
    expect(actionButton).toHaveTextContent('Cancel');

    fireEvent.click(actionButton);
    expect(cancelClickHandler).toHaveBeenCalledTimes(1);

    container = render(
      <DownloaderControls
        disableDownloadButton={false}
        downloadClickHandler={downloadClickHandler}
        clearClickHandler={clearClickHandler}
        cancelClickHandler={cancelClickHandler}
        downloadQueue={[
          {
            status: 'cancelled',
            url: 'https://www.youtube.com/watch?v=1',
          },
        ]}
      />
    ).container;
    actionButton = container.querySelector('button') as HTMLButtonElement;
    expect(actionButton).toHaveTextContent('Clear');
    fireEvent.click(actionButton);
    expect(clearClickHandler).toHaveBeenCalledTimes(1);
  });

  it('can disable download button', () => {
    const { container } = render(
      <DownloaderControls
        disableDownloadButton
        downloadClickHandler={jest.fn}
        clearClickHandler={jest.fn}
        cancelClickHandler={jest.fn}
        downloadQueue={[]}
      />
    );
    expect(container.querySelector('button')).toBeDisabled();
  });
});
