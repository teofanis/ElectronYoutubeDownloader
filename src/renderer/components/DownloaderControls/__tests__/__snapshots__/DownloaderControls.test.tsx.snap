// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DownloaderControls Test should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex justify-center mt-10"
      >
        <button
          class="
       rounded-lg
       font-semibold py-2 px-4 inline-flex items-center cursor-pointer
       transition duration-500 ease-in-out transform hover:-translate-y-1 focus:outline-none
       hover:shadow-lg
       focus:shadow-outline
       focus:ring-2
       ring-offset-current ring-offset-2
      ring-gray-700
      disabled:opacity-50 disabled:cursor-not-allowed
      bg-gray-300
      hover:bg-gray-100"
        >
          <svg
            class="w-4 h-4 mr-2"
            viewBox="0 0 20 20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M13 8V2H7v6H2l8 8 8-8h-5zM0 18h20v2H0v-2z"
            />
          </svg>
          <span>
            Download
          </span>
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex justify-center mt-10"
    >
      <button
        class="
       rounded-lg
       font-semibold py-2 px-4 inline-flex items-center cursor-pointer
       transition duration-500 ease-in-out transform hover:-translate-y-1 focus:outline-none
       hover:shadow-lg
       focus:shadow-outline
       focus:ring-2
       ring-offset-current ring-offset-2
      ring-gray-700
      disabled:opacity-50 disabled:cursor-not-allowed
      bg-gray-300
      hover:bg-gray-100"
      >
        <svg
          class="w-4 h-4 mr-2"
          viewBox="0 0 20 20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M13 8V2H7v6H2l8 8 8-8h-5zM0 18h20v2H0v-2z"
          />
        </svg>
        <span>
          Download
        </span>
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
